cmake/configure/configure_40_cuda.cmake

```diff
@@ -0,0 +1,124 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2016 - 2022 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE.md at
+## the top level directory of deal.II.
+##
+## ---------------------------------------------------------------------
+
+#
+# Configuration for cuda support:
+#
+
+#
+# cuda support is experimental. Therefore, disable the feature per default:
+#
+SET(DEAL_II_WITH_CUDA FALSE CACHE BOOL "")
+
+MACRO(FEATURE_CUDA_FIND_EXTERNAL var)
+  IF(NOT Kokkos_ENABLE_CUDA)
+    SET(CUDA_ADDITIONAL_ERROR_STRING
+      ${CUDA_ADDITIONAL_ERROR_STRING}
+      "deal.II can only be compiled with Cuda support if Kokkos was built with Cuda support!"
```

Comment:
Maybe it's worth mentioning if `DEAL_II_WITH_TRILINOS="ON" `, then Trilinos needs to be built with CUDA support?